gobblin-salesforce/src/main/java/org/apache/gobblin/typedconfig/ConstraintUtil.java

```diff
@@ -0,0 +1,80 @@
+package org.apache.gobblin.typedconfig;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import org.apache.gobblin.typedconfig.compiletime.EnumOptions;
+import org.apache.gobblin.typedconfig.compiletime.IntRange;
+import org.apache.gobblin.typedconfig.compiletime.LongRange;
+import org.apache.gobblin.typedconfig.compiletime.StringRegex;
+
+
+public class ConstraintUtil {
+  private ConstraintUtil() {
+  }
+
+  public static Object constraint(Field field, Object value, Object defaultValue) {
+    if (value == null) {
+      return defaultValue;
+    }
+    Class type = field.getType();
+    if (type == int.class) {
```

Comment:
too many `if...else`, use switch and `FieldValidator` subclasses?

Reply:
Better keep using if-else. I think `switch` is like for the case there is very little code between  switch-cases. https://docs.oracle.com/javase/8/docs/technotes/guides/language/strings-switch.html

Reply:
I mean something like this https://github.com/apache/incubator-gobblin/blob/master/gobblin-modules/gobblin-parquet/src/main/java/org/apache/gobblin/converter/parquet/JsonElementConversionFactory.java.

Reply:
create separate function for every type of data.